bpf/bpf_lxc.c

```diff
@@ -829,17 +829,16 @@ static inline int __inline__ ipv6_policy(struct __sk_buff *skb, int ifindex, __u
 			return ret2;
 	}
 
-	/* Policy lookup is done on every packet to account for packets that
-	 * passed through the allowed consumer. */
-	/* FIXME: Add option to disable policy accounting and avoid policy
-	 * lookup if policy accounting is disabled */
 	verdict = policy_can_access(&POLICY_MAP, skb, src_label, tuple.dport,
 				    tuple.nexthdr, sizeof(tuple.saddr),
 				    &tuple.saddr);
-	if (unlikely(ret == CT_NEW)) {
-		if (verdict != TC_ACT_OK)
-			return DROP_POLICY;
 
+	/* Reply packets and related packets are allowed, all others must be
+	 * permitted by policy */
+	if (ret != CT_REPLY && ret != CT_RELATED && verdict != TC_ACT_OK)
```

Comment:
Interesting subtle change here that means that we no longer rely on CT for established connections; such packets now rely on policy. Makes sense, means we're less reliant on the CT state.

Reply:
Seems `if ret = CT_ESTABLISHED` the policy_can_access could be skipped and we could jump to the proxy port check directly after ct_lookup(). I'm fine if that goes in as an optimization later but could save some map lookups.

Reply:
nevermind re-reading code its clear the above comment is not correct. But do I read this correct, if ct_lookup is CT_REPLY or CT_RELATED then we will return 0 sans NAT requirements.